include/for_each.hpp

```diff
@@ -0,0 +1,113 @@
+#pragma once
+
+#include <utility>
+
+#include "ddc/chunk_span.hpp"
+#include "ddc/discrete_coordinate.hpp"
+#include "ddc/discrete_domain.hpp"
+
+/// Serial execution
+struct serial_policy
+{
+};
+
+/** iterates over a 1d domain
+ * @param[in] domain the 1d domain to iterate
+ * @param[in] f      a functor taking an index as parameter
+ */
+template <class DDim, class Functor>
+inline void for_each(serial_policy, DiscreteDomain<DDim> const& domain, Functor&& f) noexcept
+{
+    for (DiscreteCoordinate<DDim> const i : domain) {
+        f(i);
+    }
+}
+
+/** iterates over a 2d domain
+ * @param[in] domain the 2d domain to iterate
+ * @param[in] f      a functor taking 2 indices as parameter
+ */
+template <class DDim1, class DDim2, class Functor>
+inline void for_each(
+        serial_policy,
+        DiscreteDomain<DDim1, DDim2> const& domain,
+        Functor&& f) noexcept
+{
+    for (DiscreteCoordinate<DDim1> const i1 : select<DDim1>(domain)) {
+        for (DiscreteCoordinate<DDim1> const i2 : select<DDim2>(domain)) {
+            f(DiscreteCoordinate<DDim1, DDim2>(i1, i2));
+        }
+    }
+}
+
+/** OpenMP parallel execution
+ * - default scheduling
+ * - collapsing in case of tightly nested loops
+ */
+struct omp_policy
```

Comment:
you could create a constexpr instance for the users to use

Reply:
Done